pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -202,19 +242,52 @@ func (configgen *ConfigGeneratorImpl) BuildDeltaClusters(proxy *model.Proxy, pus
 		model.XdsLogDetails{AdditionalInfo: fmt.Sprintf("cached:%v/%v", clusterCacheStats.hits, clusterCacheStats.hits+clusterCacheStats.miss)}, true
 }
 
+// getRemovedClusterName finds the name of the removed cluster given a hostname and list of cluster names.
+// subsets is ignored if it is nil
+func getRemovedClusterNames(resNames []string, subsets []string, target string) []string {
+	removed := make([]string, 0)
+	for _, n := range resNames {
+		_, sub, svcHost, _ := model.ParseSubsetKey(n)
+		if svcHost == host.Name(target) && (len(subsets) == 0 || contains(subsets, sub)) {
```

Comment:
As for a dr deleted, it satisfies `len(subsets) == 0`, if a cluster svcHost = dr.Host, the cluster will be removed. 

Think about the service maybe still there, in this case envoy will miss this cluster.

Reply:
so if a destinationRule is deleted, do we send an update instead of delete @hzxuzhonghu?

Reply:
I think if a subset is deleted, we need to send the subset cluster deletion

Reply:
Istio support destinationrules merge, let's say subset v1 can be in dr1, and subset v2 can be in dr2.

Reply:
not really sure what you mean, could you elaborate?

Reply:
Please take a look at PushContext.PushContext

There is a case, more than one destinationRule can be defined for a single service, though this is not common


```
// apiVersion: networking.istio.io/v1alpha3
// kind: DestinationRule
// metadata:
//   name: bookinfo-ratings
// spec:
//   host: ratings.prod.svc.cluster.local
//   trafficPolicy:
//     loadBalancer:
//       simple: LEAST_CONN
//   subsets:
//   - name: testversion
//     labels:
//       version: v3
//     trafficPolicy:
//       loadBalancer:
//         simple: ROUND_ROBIN


---

// apiVersion: networking.istio.io/v1alpha3
// kind: DestinationRule
// metadata:
//   name: bookinfo-ratings-v2
// spec:
//   host: ratings.prod.svc.cluster.local
//   trafficPolicy:
//     loadBalancer:
//       simple: LEAST_CONN
//   subsets:
//   - name: testversion
//     labels:
//       version: v2
//     trafficPolicy:
//       loadBalancer:
//         simple: ROUND_ROBIN
```